### PR TITLE
Deactivate BlochainParameters Contract on L2

### PR DESCRIPTION
**Pull request by soloseng**
### Description

Functions that call precompiles now revert on L2.

### Other changes

Removed an edge-case test that was only relevant when running test on Ganache, where the epoch size was set to 100

### Tested

Unit testes

### Related issues

- Fixes https://github.com/celo-org/celo-blockchain-planning/issues/192

### CODE REVIEW

File: packages/protocol/test-sol/governance/network/BlockchainParameters.t.sol [hunk 1]
```diff
@@ -8,11 +8,10 @@ import { Constants } from "@test-sol/constants.sol";
 import { Utils } from "@test-sol/utils.sol";
 
 contract BlockchainParametersTest is Test, Constants, Utils {
-  // using the mainnet epoch size would not allow to test an edge case
-  uint256 constant EPOCH_SIZE = 100;
```
Comment (arthurgousset) [inline packages/protocol/test-sol/governance/network/BlockchainParameters.t.sol:12]: Noting that the `EPOCH_SIZE` is being set to 100 in this PR:
- https://github.com/celo-org/celo-monorepo/pull/11009

https://github.com/celo-org/celo-monorepo/blob/3e738a62bb6cea403fbecbd965926f8128a7854b/packages/protocol/test-sol/precompiles/EpochSizePrecompile.sol#L7

**Review by arthurgousset — COMMENTED** [review]
Left a small comment on `EPOCH_SIZE`. Not sure if any changes are needed, but thought I'd flag in case.

File: packages/protocol/contracts/governance/BlockchainParameters.sol [hunk 2]
```diff
@@ -5,10 +5,12 @@ import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
 import "../common/Initializable.sol";
 import "../common/UsingPrecompiles.sol";
 
+import "../../contracts-0.8/common/IsL2Check.sol";
+
 /**
  * @title Contract for storing blockchain parameters that can be set by governance.
```
Comment (pahor167) [inline packages/protocol/contracts/governance/BlockchainParameters.sol:11]: setBlockGasLimit and setIntrinsicGasForAlternativeFeeCurrency should also be deprecated

File: packages/protocol/test-sol/governance/network/BlockchainParameters.t.sol (hunk 1 above)
Comment (soloseng) [reply packages/protocol/test-sol/governance/network/BlockchainParameters.t.sol:12]: This should not be affected, since these test are running on forge. 

That being said, you might have to add some logic during [integration tests](https://github.com/celo-org/celo-monorepo/pull/11002#top) that sets the `EPOCH_SIZE` back to 100, since all forge test currently have the EPOCH_SIZE set to 17280.

File: packages/protocol/contracts/governance/BlockchainParameters.sol (hunk 2 above)
Comment (soloseng) [reply packages/protocol/contracts/governance/BlockchainParameters.sol:11]: I can add it, but it did not really seem that this was required, since these state variables are not referenced by the contract and these functions are only called by the owner.

File: packages/protocol/test-sol/governance/network/BlockchainParameters.t.sol (hunk 1 above)
Comment (arthurgousset) [reply packages/protocol/test-sol/governance/network/BlockchainParameters.t.sol:12]: That makes sense, thanks for clarifying

File: packages/protocol/contracts/governance/BlockchainParameters.sol (hunk 2 above)
Comment (pahor167) [reply packages/protocol/contracts/governance/BlockchainParameters.sol:11]: Lets please add it for consistency

Comment (soloseng) [reply packages/protocol/contracts/governance/BlockchainParameters.sol:11]: It's been added as of https://github.com/celo-org/celo-monorepo/pull/11008/commits/df404b22f36bc2c2879ac2725e4cc95079f469fb

**Review by pahor167 — APPROVED** [review]
